src/App.tsx

```diff
@@ -6,55 +1,111 @@
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+import { User } from './types/User';
+import { Post } from './types/Post';
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+import { getUsers } from './api/users';
+import { getPosts } from './api/posts';
+import 'bulma/css/bulma.css';
+import '@fortawesome/fontawesome-free/css/all.css';
 
-              <Loader />
+import './App.scss';
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userSelected, setUserSelected] = useState<User | null>(null);
+
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+  const getUsersPost = useCallback((userId: User['id']) => {
+    setError('');
+    setIsLoading(true);
+
+    getPosts(userId)
+      .then(setPosts)
+      .catch(() => setError('Unable to load posts!'))
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    setError('');
+
+    getUsers()
+      .then(setUsers)
+      .catch(() => setError('Unable to load users!'));
+  }, []);
+
+  return (
+    <main className="section">
+      <div className="container">
+        <div className="tile is-ancestor">
+          <div className="tile is-parent">
+            <div className="tile is-child box is-success">
+              <div className="block">
+                <UserSelector
+                  users={users}
+                  userSelected={userSelected}
+                  setUserSelected={setUserSelected}
+                  getUsersPost={getUsersPost}
+                />
               </div>
 
-              <PostsList />
+              <div className="block" data-cy="MainContent">
+                {!userSelected && !error && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
+
+                {isLoading && <Loader />}
+
+                {error && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    {error}
+                  </div>
+                )}
+
+                {userSelected && !error && posts.length === 0 && !isLoading && (
```

Comment:
The condition `userSelected && !error && posts.length === 0 && !isLoading` might be complex. Ensure that each part of this condition is necessary and correctly reflects the intended logic for displaying the 'No posts yet' message.